### PR TITLE
Major rewrite of documentation.

### LINKED ISSUE #1: Rewrite the documentation
After all of the recent refactoring work, much of the documentation is now plain wrong. Much was also missing for the bwa-meth aligner from day one.

### PR DESCRIPTION
**Pull request by ewels**
Fixes #1

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -26,101 +31,92 @@ mv nextflow ~/bin
...
 
-Note that these limits are the maximum to be used _per task_. Nextflow will automatically attempt to parallelise as many jobs as possible given the available resources.
+If running offline with Singularity, you'll need to download and transfer the Singularity image first:
 
-## 3.2) Configuration: Docker and Singularity
-Running the pipeline with the option `-with-docker` tells Nextflow to enable docker for this run. The above base profile already specifies the container to be used (hub.docker.com/r/nf-core/methylseq).
-
-If you don't have Docker available and do have [Singularity](http://singularity.lbl.gov/), instead use `-with-singularity` on the command line.
+```bash
+# Switch 1.4 with the version of the pipeline that you're using
```
Comment (MaxUlysse) [inline docs/installation.md]: I would directy use something like [VERSION] that way, it's pretty clear that you have to change version, and the doc is always up to date ;-)

File: docs/installation.md [hunk 2]
```diff
@@ -26,101 +31,92 @@ mv nextflow ~/bin
...
-
-If you don't want to use the illumina iGenomes you can supply either a Bismark reference or a FASTA file. If a Bismark reference is specified, the pipeline won't have to generate it and will be finished quite a bit faster. If a FASTA file is supplied then the Bismark reference will be built when the pipeline starts. Use the command line option `--saveReference` to keep the generated references so that they can be added to your config and used again in the future. Use  `--bismark_index` or `--fasta` to specify the paths to the reference.
-
-Alternatively, you can add the paths to your NextFlow config under a relevant id and just specify this id with `--genome ID` when you run the pipeline:
+Alternatively, ou can supply either a Bismark / BWA reference or a FASTA file. This can be done on the command line (see the [usage docs](usage.md#supplying-reference-indices)).
+You can use the same pipeline config framework as used with iGenomes to specify multiple referenes of your own. Add the paths to your NextFlow config under a relevant id and just specify this id with `--genome ID` when you run the pipeline:
 
 ```groovy
```
Comment (MaxUlysse) [inline docs/installation.md]: I'm guessing there's no Nextflow in Markdown yet?

Comment (ewels) [reply docs/installation.md]: There is - it uses the same linguist (?) parsing that the rest of the site uses.

Comment (MaxUlysse) [reply docs/installation.md]: Can we specify ```nextflow then?

Comment (ewels) [reply docs/installation.md]: Yup!

**Comment by MaxUlysse** [comment]
Quite impressive.
We're planning some improvements on Sarek documentation as well.
I'll be sure to follow such a good example ;-)

**Comment by ewels** [comment]
Thanks! I'm hoping to use much of this text in the cookiecutter template once merged. Then across to the RNA pipeline, etc..

File: docs/installation.md [hunk 3]
```diff
@@ -131,3 +127,19 @@ params {
...
+## Appendices
+
+#### Running on UPPMAX
+To run the pipeline on the [Swedish UPPMAX](https://www.uppmax.uu.se/) clusters (`rackham`, `irma`, `bianca` etc), use the command line flag `-profile uppmax`. This tells Nextflow to submit jobs using the SLURM job executor with Singularity for software dependencies.
+
+Note that you will need to specify your UPPMAX project ID when running a pipeline. To do this, use the command line flag `--project <project_ID>`. The pipeline will exit with an error message if you try to run it pipeline with the default UPPMAX config profile without a project.
+
+**Optional Extra:** To avoid having to specify your project every time you run Nextflow, you can add it to your personal Nextflow config file instead. Add this line to `~/.nextflow/config`:
+
+```groovy
```
Comment (MaxUlysse) [inline docs/installation.md]: There is another ```groovy

File: README.md [hunk 4]
```diff
@@ -9,17 +9,30 @@
...
 
 The pipeline uses [Nextflow](https://www.nextflow.io), a bioinformatics workflow tool. It pre-processes raw data from FastQ inputs, aligns the reads and performs extensive quality-control on the results.
 
-### Choice of workflows
+### Pipeline Steps
+
+The pipeline allows you to choose between running either [Bismark](https://github.com/FelixKrueger/Bismark) or [bwa-meth](https://github.com/brentp/bwa-meth) / [MethylDackel](https://github.com/dpryan79/methyldackel).
+Choose between workflows by using `--aligner bismark` (default) or `--aligner bwameth`.
+
+| Step   | Bismark Workflow | bwa-meth Workflow |
```
Comment (MaxUlysse) [inline README.md]: Just one last comment, table doesn't look good in plain text, but I guess it's difficult to align everything

Comment (ewels) [reply README.md]: Yeah, I'll redo it now. I was offline when I wrote this but can redo it using an online tool, then it's easy.

**Pull request merged**